src/gurt/tests/test_gurt.c

```diff
@@ -710,7 +710,7 @@ test_log(void **state)
 	setenv("D_LOG_MASK", "T2=WARN", 1);
 	setenv("DD_MASK", "trace", 1);
 	d_log_sync_mask();
-	D_STRNDUP(logmask, "T2=WARN", 32);
+	D_STRNDUP(logmask, "T2=WARN", 8);
```

Comment:
why not D_STRNDUP_S ?

Reply:
Because in this (and other cases) the length of the string being copied is the same length as a pointer so the _Static_assert() is failing.

It would work, and it would be correct, but we'd have to remove the static assert and that would leave the macro open to mis-use.